### PR TITLE
[MSYS-777] Added Azure CLI 2.0 support.

### PR DESCRIPTION
**Pull request by piyushawasthi**
**Description:**
Added Azure CLI 2.0 support and re-factor code for following.
- Update azurerm server create
- Update azurerm server list
- Update azurerm server show
- Update azurerm server delete

**Reason:**
Microsoft recently released the Azure CLI 2.0 and the commands start with “az” instead of “azure” like Azure CLI 1.0.
Ref: https://buildazure.com/2017/05/16/azure-cli-1-0-vs-2-0-compared-installation-and-usage/

**Tested:**
- Verified azurerm server create command on both Azure CLI 1.0 and Azure CLI 2.0
- Verified azurerm server list command on both Azure CLI 1.0 and Azure CLI 2.0
- Verified azurerm server show command on both Azure CLI 1.0 and Azure CLI 2.0
- Verified azurerm server delete command on both Azure CLI 1.0 and Azure CLI 2.0

**Resolved Issue:**
knife azurerm command returns "azure: not recognized as an internal or external command".
Ref: https://github.com/chef/knife-azure/issues/466

Signed-off-by: piyushawasthi <piyush.awasthi@msystechnologies.com>

### CODE REVIEW

**Comment by piyushawasthi** [comment]
Yes, I am working on that.

**Comment by piyushawasthi** [comment]
Hi @btm , @Vasu1105 I have done with that card please review.

File: lib/chef/knife/azurerm_base.rb [hunk 1]
```diff
@@ -141,56 +136,49 @@ def is_token_valid?(token_details)
           return false
         elsif time_difference <= 600 # 600sec = 10min
           # This is required otherwise a long running command may fail inbetween if the token gets expired.
-          raise "Token will expire within 10 minutes. Please run 'azure login' command"
+          raise_azure_timeout
```
Comment (Vasu1105) [inline lib/chef/knife/azurerm_base.rb]: I guess we should only make method only if there is a code repetition.

Comment (piyushawasthi) [reply lib/chef/knife/azurerm_base.rb]: done

**Comment by piyushawasthi** [comment]
Thanks a lot @btm

**Pull request merged**